src/charm.py

```diff
@@ -35,6 +43,23 @@
 ]
 
 
+class SpeakerConfig(BaseModel):
+    name: str = Field(
+        ...,
+        regex="^[a-z0-9.-]+$",
```

Comment:
ok, so lowercase + numbers and period and dash with at least one character

Reply:
yes that's correct. case-wise the name field is restricted to lower case, but the node selector can have upper and lower case characters.